pkg/registry/core/service/ipallocator/ipallocator_test.go

```diff
@@ -587,6 +587,18 @@ func Test_addOffsetAddress(t *testing.T) {
 			offset:  128,
 			want:    netip.MustParseAddr("192.168.0.128"),
 		},
+		{
+			name:    "IPv4 with leading zeros",
+			address: netip.MustParseAddr("0.0.1.8"),
+			offset:  138,
+			want:    netip.MustParseAddr("0.0.1.146"),
+		},
+		{
+			name:    "IPv6 with leading zeros",
+			address: netip.MustParseAddr("fc::1"),
```

Comment:
should we denote this as "00fc" to make it clear?  It took me a minute to figure out what was missing here

Reply:
+1 from me. That `00` prefix is what lead to the issue. So, unless there is some standard of how the IPs are denoted that prevents it, it would be nice to have `00` prefixed.

Reply:
agree, the problem of this bug is this subtle details, 00fc vs fc00

Reply:
done